network/discovery/observability.go

```diff
@@ -0,0 +1,59 @@
+package discovery
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+
+	"github.com/ssvlabs/ssv/observability"
+)
+
+const (
+	observabilityName      = "github.com/ssvlabs/ssv/network/discovery"
+	observabilityNamespace = "ssv.p2p.discovery"
+)
+
+type rejectionReason string
+
+const (
+	reachedLimitReason    rejectionReason = "reachedLimit"
+	noSharedSubnetsReason rejectionReason = "noSharedSubnets"
+	zeroSubnetsReason     rejectionReason = "zeroSubnets"
+)
+
```

Comment:
names here are kinda confusing, because we just rejection in message validation and also connections I guess. this sounds like we are rejecting connections or something, but its a much more lightweight action here, we are filtering/skipping peers that we found in discovery and aren't fitting for us. we are not connected or rejecting connections with us. 

maybe a better terminology will be skipping or filter?

Reply:
Following the naming in the code, I see that mostly word "skip" is used. I will refactor the metric

Reply:
Renamed the metric from `ssv.p2p.discovery.peers.rejected` to `ssv.p2p.discovery.peers.skipped`, also renamed attribute from `ssv.p2p.discovery.rejection_reason` to `ssv.p2p.discovery.skip_reason`